pkg/target/target_integration_test.go

```diff
@@ -0,0 +1,300 @@
+			ns:         makeNamespace("my-ns"),
+			constraint: makeConstraint(setLabelSelector("different", "label")),
+			allowed:    true,
+		},
+		{
+			name:       "match nsselector",
+			obj:        makeResource("some", "Thing"),
+			ns:         makeNamespace("my-ns", map[string]string{"a": "label"}),
+			constraint: makeConstraint(setNamespaceSelector("a", "label")),
+			allowed:    false,
+		},
+		{
+			name:       "no match nsselector",
+			obj:        makeResource("some", "Thing"),
+			ns:         makeNamespace("my-ns", map[string]string{"a": "label"}),
+			constraint: makeConstraint(setNamespaceSelector("different", "label")),
+			allowed:    true,
+		},
+		{
+			name:       "match kinds",
+			obj:        makeResource("some", "Thing"),
+			ns:         makeNamespace("my-ns"),
+			constraint: makeConstraint(setKinds([]string{"some"}, []string{"Thing"})),
+			allowed:    false,
+		},
+		{
+			name:       "no match kinds",
+			obj:        makeResource("some", "Thing"),
+			ns:         makeNamespace("my-ns"),
+			constraint: makeConstraint(setKinds([]string{"different"}, []string{"Thing"})),
+			allowed:    true,
+		},
+		{
+			name: "match everything",
+			obj:  makeResource("some", "Thing", map[string]string{"obj": "label"}),
+			ns:   makeNamespace("my-ns", map[string]string{"ns": "label"}),
+			constraint: makeConstraint(
+				setKinds([]string{"some"}, []string{"Thing"}),
+				setNamespaceName("my-ns"),
+				setLabelSelector("obj", "label"),
+				setNamespaceSelector("ns", "label"),
+			),
+			allowed: false,
+		},
+		{
+			name: "match everything but kind",
+			obj:  makeResource("some", "Thing", map[string]string{"obj": "label"}),
+			ns:   makeNamespace("my-ns", map[string]string{"ns": "label"}),
+			constraint: makeConstraint(
+				setKinds([]string{"different"}, []string{"Thing"}),
+				setNamespaceName("my-ns"),
+				setLabelSelector("obj", "label"),
+				setNamespaceSelector("ns", "label"),
+			),
+			allowed: true,
+		},
+		{
+			name: "match everything but namespace",
+			obj:  makeResource("some", "Thing", map[string]string{"obj": "label"}),
+			ns:   makeNamespace("my-ns", map[string]string{"ns": "label"}),
+			constraint: makeConstraint(
+				setKinds([]string{"some"}, []string{"Thing"}),
+				setNamespaceName("different-ns"),
+				setLabelSelector("obj", "label"),
+				setNamespaceSelector("ns", "label"),
+			),
+			allowed: true,
+		},
+		{
+			name: "match everything but labelselector",
+			obj:  makeResource("some", "Thing", map[string]string{"obj": "label"}),
+			ns:   makeNamespace("my-ns", map[string]string{"ns": "label"}),
+			constraint: makeConstraint(
+				setKinds([]string{"some"}, []string{"Thing"}),
+				setNamespaceName("my-ns"),
+				setLabelSelector("obj", "different-label"),
+				setNamespaceSelector("ns", "label"),
+			),
+			allowed: true,
+		},
+		{
+			name: "match everything but nsselector",
+			obj:  makeResource("some", "Thing", map[string]string{"obj": "label"}),
```

Comment:
nit: maybe add one test case for oldObject here?

Reply:
Added test for oldObject, which led me to add handling in the target handler for the case where "object" in the review is null that sources the object-under-test from oldObject.